crates/orderbook/src/account_balances.rs

```diff
@@ -215,6 +227,32 @@ impl BalanceFetching for Web3BalanceFetcher {
         };
         Ok(success)
     }
+
+    async fn has_sufficient_balance(&self, token: H160, from: H160, amount: U256) -> Result<bool> {
```

Comment:
In theory, this also needs to pass in the `SellTokenSource` because we would need to query the `Vault::internal_balance` in the case of `SellTokenSource::Internal`.